File: schemaorggo/techarticle.go
package schemaorggo

import "encoding/json"

// TechArticle see : https://schema.org/TechArticle
type TechArticle struct {
	Article

	typeContext

	// Dependencies see : https://schema.org/dependencies
	// Prerequisites needed to fulfill steps in article.
	Dependencies string `json:"dependencies,omitempty"`

	// ProficiencyLevel see : https://schema.org/proficiencyLevel
	// Proficiency needed for this content; expected values: 'Beginner', 'Expert'.
	ProficiencyLevel string `json:"proficiencyLevel,omitempty"`
}

func (v TechArticle) MarshalJSONWithTypeContext() ([]byte, error) {
	v.C = "http://schema.org"
	v.T = "TechArticle"

	return json.Marshal(v)
}

func (v *TechArticle) MarshalJSON() ([]byte, error) {
	if v == nil {
		return []byte("null"), nil
	}

	v.C = "http://schema.org"
	v.T = "TechArticle"

	return json.Marshal(*v)
}

File: schemaorgphp/classes/tech_article.php
<?php

class TechArticle extends Article implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'TechArticle';
	
	/**
	 * Prerequisites needed to fulfill steps in article.
	 * see : https://schema.org/dependencies
	 * @var string|string[]
	 */
	public var $dependencies;
	
	/**
	 * Proficiency needed for this content; expected values: &#39;Beginner&#39;, &#39;Expert&#39;.
	 * see : https://schema.org/proficiencyLevel
	 * @var string|string[]
	 */
	public var $proficiency_level;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'TechArticle'
		);
		
		$serialized = so_json_serialize( $this->dependencies );
		if ( ! empty( $serialized ) ) {
			$out['dependencies'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->proficiency_level );
		if ( ! empty( $serialized ) ) {
			$out['proficiencyLevel'] = $serialized;
		}
		
		return $out;
	}
}
